core/dbt/include/global_project/macros/adapters/timestamps.sql

```diff
@@ -0,0 +1,62 @@
+{%- macro current_timestamp() -%}
+    {{ adapter.dispatch('current_timestamp', 'dbt')() }}
+{%- endmacro -%}
+
+{% macro default__current_timestamp() %}
+    current_timestamp
+{% endmacro %}
+
+{%- macro current_timestamp_in_utc() -%}
+    {{ adapter.dispatch('current_timestamp_in_utc', 'dbt')() }}
+{%- endmacro -%}
+
+{% macro default__current_timestamp_in_utc() %}
+    {{ convert_timezone(target_tz="UTC",
+        source_timestamp=current_timestamp())}}
+{% endmacro %}
```

Comment:
@jtcohen6

From a product perspective, what do you want the return type to be for this version of `current_timestamp_in_utc`?

Let's suppose for simplicity that all data platforms have a timestamp data types with equivalent semantics to those defined in the SQL standard \*:
- `timestamp without time zone`
- `timestamp with time zone`

Then there are three options for the data type produced by `current_timestamp_in_utc`:
1. `timestamp without time zone` for all adapters
1. `timestamp with time zone` for all adapters
1. a mix of return types across adapters -- some with one type and some with the other

---

\* Although extremely uncommon, there are some platforms that only support a single timestamp data type (e.g., Spark and Databricks).

Reply:
Opened a new issue to answer this thread, and reflect our discussion earlier today: https://github.com/dbt-labs/dbt-core/issues/5969